### PR TITLE
lib/core: add Sequence::modulo to access with a Python-wrap semantic

### PR DESCRIPTION
**Pull request by privat**
Proposed because https://github.com/nitlang/nit/pull/1920/files#diff-c6e41a937b2b8ba5e5e04bd8f49260f7R104

I'm not sure of the name `modulo`.

### CODE REVIEW

File: lib/core/collection/abstract_collection.nit [hunk 1]
```diff
@@ -812,6 +812,42 @@ interface SequenceRead[E]
 	# REQUIRE `index >= 0 and index < length`
 	fun [](index: Int): E is abstract
 
+	# Return the index-th element but wrap
+	#
+	# Whereas `self[]` requires the index to exists, the `modulo` accessor automatically
+	# wraps overbound and underbouds indexess.
```
Comment (xymus) [inline lib/core/collection/abstract_collection.nit]: There's one too many s in `indexess`

File: lib/core/collection/abstract_collection.nit [hunk 2]
```diff
@@ -1062,6 +1098,23 @@ interface Sequence[E]
 	# REQUIRE `index >= 0 and index <= length`
 	fun []=(index: Int, item: E) is abstract
 
+	# Set the index-th element but wrap
+	#
+	# Whereas `self[]i=` requires the index to exists, the `modulo` accessor automatically
+	# wraps overbound and underbouds indexess.
```
Comment (xymus) [inline lib/core/collection/abstract_collection.nit]: There's an `i` lost in `self[]i=` and the extra s in `indexess`

File: lib/core/collection/abstract_collection.nit [hunk 3]
```diff
@@ -812,6 +812,42 @@ interface SequenceRead[E]
...
+	# ~~~
+	#
+	# REQUIRE `not_empty`
+	fun modulo_index(index: Int): Int
+	do
+		var l = length
+		if index >= 0 then
+			return index % l
+		else
+			return (l + index) % l
```
Comment (xymus) [inline lib/core/collection/abstract_collection.nit]: `+ length` (or `l +` here) is not enough to support large negative values.

**Comment by xymus** [comment]
As an alternative to `modulo`, why not `wrap`?

File: lib/core/collection/abstract_collection.nit (hunk 3 above)
Comment (xymus) [inline lib/core/collection/abstract_collection.nit]: It should be the other way around `(index % l) + l` if I'm not mistaken.

**Comment by privat** [comment]
reroll: support large negative values

I do not like `wrap` either, and it seems to me less POLA since `arr.wrap(4)` might be understood like some circular shifting operation.

**Comment by xymus** [comment]
+1

**Pull request merged**